src/routerify.ts

```diff
@@ -0,0 +1,36 @@
+import xs from 'xstream';
+import {adapt} from '@cycle/run/lib/adapt';
+import {RouteMatcher} from './interfaces';
+import {RouterSource} from './RouterSource';
+import {Location, createPath} from 'history';
+import {HistoryInput,  GenericInput} from '@cycle/history';
+import {Stream} from 'xstream';
+
+export declare type HistoryAction = HistoryInput | GenericInput | string;
+export declare type RouterSink = Stream<HistoryAction>;
+
+/**
+ * Wraps main to provide an advanced interface over @cycle/history
+ * @public
+ * @method routerify
+ * @return {main} The augmented main function
+ */
+function routerify(main: (a: any) => any, routeMatcher: RouteMatcher, basename = '/', historyName = 'history', routerName = 'router') {
+    if (typeof main !== 'function') {
+        throw new Error('First argument to routerify must be a valid cycle app');
+    }
+    const createHref = (location: Location) => basename + createPath(location);
+    return function(sources: any): any {
```

Comment:
I can see arguments for both sides... and though I prefer hiding the `history` by default, if you think it will ease history -> router transitions having `history` exposed by default then I'm okay with that too...